lib/resty/kafka/ringbuffer.lua

```diff
@@ -46,6 +75,19 @@ function _M.add(self, topic, key, message)
 end
 
 
+function _M.to_post(self)
```

Comment:
available might be a better function name?

Reply:
> available might be a better function name?

how about to use try_release_buffer_wait?